pkg/operator/ceph/csi/spec.go

```diff
@@ -470,14 +473,57 @@ func startDrivers(namespace string, clientset kubernetes.Interface, ver *version
 	}
 
 	if ver.Major > KubeMinMajor || (ver.Major == KubeMinMajor && ver.Minor >= provDeploymentSuppVersion) {
-		err = createCSIDriverInfo(clientset, RBDDriverName, ownerRef)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create CSI driver object for %q", RBDDriverName)
+		if EnableRBD {
+			err = createCSIDriverInfo(clientset, RBDDriverName, ownerRef)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create CSI driver object for %q", RBDDriverName)
+			}
 		}
-		err = createCSIDriverInfo(clientset, CephFSDriverName, ownerRef)
-		if err != nil {
-			return errors.Wrapf(err, "failed to create CSI driver object for %q", CephFSDriverName)
+		if EnableCephFS {
+			err = createCSIDriverInfo(clientset, CephFSDriverName, ownerRef)
+			if err != nil {
+				return errors.Wrapf(err, "failed to create CSI driver object for %q", CephFSDriverName)
+			}
+		}
+	}
+	return nil
+}
+
+func stopDrivers(namespace string, clientset kubernetes.Interface, ver *version.Info) error {
+	if !EnableRBD {
+		if err := k8sutil.DeleteDaemonset(clientset, namespace, "csi-rbdplugin"); err != nil {
+			logger.Errorf("failed to delete the rbd plugin. %v", err)
+		}
+		if err := k8sutil.DeleteDeployment(clientset, namespace, "csi-rbdplugin-provisioner"); err != nil {
+			logger.Errorf("failed to delete the rbd plugin provisioner. %v", err)
+		}
+
+		if err := k8sutil.DeleteService(clientset, namespace, "csi-rbdplugin-metrics"); err != nil {
+			logger.Errorf("failed to delete the service. %v", err)
 		}
+
+		if err := deleteCSIDriverInfo(clientset, CephFSDriverName); err != nil {
+			logger.Errorf("failed to delete the Driver Info. %v", err)
+		}
+		logger.Infof("successfully disabled Ceph RBD driver")
```

Comment:
This seems wrong even if there is an error we are logging success at last and also `Infof` is not required as we are not formatting anything

Reply:
> This seems wrong even if there is an error we are logging success at last and also `Infof` is not required as we are not formatting anything

According to [this comment](https://github.com/rook/rook/pull/5605#discussion_r438300139) we are not returning the errors and we have log after disabling the CSI driver.

Reply:
we should log success if something has not failed. or else it should it not a success.